File: tv-series-fetcher/src/main/java/mahat/aviran/tvseriesfetcher/entities/raw_request_entities/TvSeason.java
package mahat.aviran.tvseriesfetcher.entities.raw_request_entities;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.*;

import java.util.List;

@Getter @Setter @ToString @EqualsAndHashCode @AllArgsConstructor @NoArgsConstructor
@JsonIgnoreProperties(ignoreUnknown = true)
public class TvSeason {

    @JsonProperty(value = "air_date")
    private String airDate;
    private List<TvEpisode> episodes;
    private String name;
    private String overview;
    private String id;
    @JsonProperty(value = "poster_path")
    private String posterPath;
    @JsonProperty(value = "season_number")
    private String seasonNumber;

    public String getPosterPath() {
        return this.posterPath == null ?
                null :
                String.join("", "https://image.tmdb.org/t/p/w185", this.posterPath);
    }
}

File: client/src/app/main/shared/models/tv-season.ts
import {TvEpisode} from "./tv-episode";
import {WatchlistStatus} from "./watchlist-status";

export interface TvSeason {
  discriminator: 'season';
  id: string;
  name: string;
  seasonNumber: string;
  airDate: string; // TODO
  overview: string;
  posterPath: string;
  episodes: TvEpisode[];
}

export interface WatchlistTvSeason extends TvSeason {
  watchlistStatus: WatchlistStatus
}

export function isSeason(object: any): object is WatchlistTvSeason {
  return object.discriminator === 'season';
}
